src/distances/qgram.jl

```diff
@@ -80,9 +78,141 @@ function _count(s1, s2)
 	return values(d)
 end
 
+# Turn a sequence of qgrams to a count dict for them, i.e. map each
+# qgram to the number of times it has been seen.
+function countdict(qgrams)
+    d = Dict{eltype(qgrams), Int32}()
+    for qg in qgrams
+        index = Base.ht_keyindex2!(d, qg)
+		if index > 0
+			d.age += 1
+			@inbounds d.keys[index] = qg
+			@inbounds d.vals[index] = d.vals[index][1] + 1
+		else
+			@inbounds Base._setindex!(d, 1, qg, -index)
+		end
+    end
+    d
+end
+
+abstract type AbstractQGramCounts{Q,K} end
+q(qc::AbstractQGramCounts{Q,K}) where {Q,K} = Q
+counts(qc::AbstractQGramCounts) = qc.counts
+
+struct QGramDict{Q,K} <: AbstractQGramCounts{Q,K}
+    counts::Dict{K,Int}
+end
+function QGramDict(s::Union{AbstractString, AbstractVector}, q::Integer = 2)
+    @assert q >= 1
+    qgs = qgrams(s, q)
+    QGramDict{q, eltype(qgs)}(countdict(qgs))
+end
+
+# Faster (than QgramDict) with the qgrams presorted
+struct QGramSortedArray{Q,K} <: AbstractQGramCounts{Q,K}
```

Comment:
QGramSortedVector sounds like a better name? Also, I think it should be QGrams would be better than QGram.

Reply:
Agreed, will fix.